### PR TITLE
[Intel MKL] Fix incorrect DNNL1.2 integration in pooling backprop

### PR DESCRIPTION
**Pull request by nhasabni**
This PR removes the performance degradation caused by incorrect pooling
backprop integration in Intel-MKL CPU backend.

### CODE REVIEW

File: tensorflow/core/kernels/mkl_pooling_ops_common.h [hunk 1]
```diff
@@ -64,7 +70,9 @@ struct MklPoolingParams {
         alg_kind(alg_kind),
         prop_kind(prop_kind),
         src_format(src_format),
-        src_md(src_md) {}
+        src_md(src_md),
+        diff_dst_md(diff_dst_md) {
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_pooling_ops_common.h:74]: Should this be guarded with `#ifdef ENABLE_MKLDNN_V1`?

**Review by penpornk — COMMENTED** [review]
Thank you for the PR!

File: tensorflow/core/kernels/mkl_pooling_ops_common.h (hunk 1 above)
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_pooling_ops_common.h:77]: Addressed it.

**Review by penpornk — APPROVED** [review]
Thank you very much!

**Pull request merged**